presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -254,6 +254,7 @@
     public static final String REMOVE_REDUNDANT_DISTINCT_AGGREGATION_ENABLED = "remove_redundant_distinct_aggregation_enabled";
     public static final String PREFILTER_FOR_GROUPBY_LIMIT = "prefilter_for_groupby_limit";
     public static final String PREFILTER_FOR_GROUPBY_LIMIT_TIMEOUT_MS = "prefilter_for_groupby_limit_timeout_ms";
+    public static final String USE_DEFAULTS_FOR_CORRELATED_AGGREGATIONS = "use_defaults_for_correlated_aggregations";
```

Comment:
i would give this a different name as its not very clear from a user perspective what this does.  Also, even though the most common case for the aggregation push down is correlated aggregations, the optimizer is generic.  I also wonder if this needs its own session property or if its okay to just guard it with push_aggregation_through_join.  But on the other hand, since bugs in this change could cause wrong results, maybe it is good to have its own flag.

Reply:
> i would give this a different name as its not very clear from a user perspective what this does.  

Hard to come up with a concise and descriptive name. How about "optimizer.use-defaults-for-correlated-aggregation-pushdown-through-outer-joins" and I will also do as @mbasmanova suggests and add a more detailed description to the doc

>I also wonder if this needs its own session property or if its okay to just guard it with push_aggregation_through_join. But on the other hand, since bugs in this change could cause wrong results, maybe it is good to have its own flag.

So maybe leave in the flag - but default to true?

Reply:
> Hard to come up with a concise and descriptive name. 

Naming is indeed hard. It may help to write up the documentation for the flag first. Once we have a user-friendly description we may find it easier to come up with a nice name for the flag.

Reply:
defaulting the flag to true seems reasonable to me.